stumpy/motifs.py

```diff
@@ -0,0 +1,313 @@
+
+    max_occurrences : int or None, default None
+        The maximum amount of similar occurrences to be returned. The resulting
+        occurrences are sorted by distance, so a value of `10` means that the
+        indices of the most similar `10` subsequences is returned. If `None`,
+        all occurrences in the given tolerance range are returned.
+
+    atol : float or None, default None
+        Absolute tolerance (see equation in description).
+        If `None`, defaults to `0.25 * np.std(P)`
+
+    rtol : float, default 1.0
+        Relative tolerance (see equation in description).
+
+    normalize : bool, default True
+        When set to `True`, this z-normalizes subsequences prior to computing distances.
+        Otherwise, this function gets re-routed to its complementary non-normalized
+        equivalent set in the `@core.non_normalized` function decorator.
+
+    Return
+    ------
+    top_motif_indices : ndarray
+        Array of top motif indices. There are at most `k` rows, each of which represents
+        one motif. All found occurrences within a distance of `atol + MP * rtol` (where
+        MP is the matrix profile value of the motif representative) are returned. The
+        array has either `max_neighbors` columns if specified, or as many columns as
+        number of occurrences for the motif with most found occurrences. Everything else
+        is filled up with -1.
+
+    top_motif_values : ndarray
+        For every occurrence its distance to the motif representative.
+    """
+    if k < 1:  # pragma: no cover
+        logger.warn("The number of motifs, `k`, must be greater than or equal to 1")
+        logger.warn("`k` has been set to `1`")
+        k = 1
+
+    if T.ndim != 1:  # pragma: no cover
+        raise ValueError(
+            f"T is {T.ndim}-dimensional and must be 1-dimensional. "
+            "Multidimensional motif discovery is not yet supported."
+        )
+
+    if P.ndim != 1:  # pragma: no cover
+        raise ValueError(
+            f"T is {P.ndim}-dimensional and must be 1-dimensional. "
+            "Multidimensional motif discovery is not yet supported."
+        )
+
+    T = T.copy()
+    T = T[np.newaxis, :]
+
+    P = P.copy().astype(float)
+    P = P[np.newaxis, :]
+
+    m = T.shape[1] - P.shape[1] + 1
+
+    if excl_zone is None:  # pragma: no cover
+        excl_zone = int(np.ceil(m / 4))
+    if atol is None:  # pragma: no cover
+        atol = 0.25 * np.std(P)
+    if max_occurrences is None:  # pragma: no cover
+        max_occurrences = np.inf
+
+    T, M_T, Σ_T = core.preprocess(T, m)
+
+    result = _motifs(
+        T=T,
+        P=P,
+        k=k,
+        M_T=M_T,
+        Σ_T=Σ_T,
+        excl_zone=excl_zone,
+        min_neighbors=min_neighbors,
+        max_occurrences=max_occurrences,
+        atol=atol,
+        rtol=rtol,
+        normalize=normalize,
+    )
+    result = _fill_result_with_nans(result[0], result[1])
+
+    return result
+
+
+def occurrences(
```

Comment:
I'm not big on long names but I feel like we should call this function "similarity_search" to be more obvious. What do you think?

Reply:
I'm also not the biggest fan of the current name, but `similarity_search` sounds somewhat broad. Maybe something like `similar_subsequences`?

Reply:
How about `match(Q, T)`, `match_pattern(Q, T)`? I like the simplicity of `match` which is a function that looks for matches of `Q` in `T` and has parameters to allow you to be less/more strict in how you "match"

Reply:
I would consider renaming the "Fast Pattern Searching" tutorial to "Fast Pattern Matching" too

Reply:
Let's just go with the simpler/shorter `match` then :)